src/VisualStudio/IntegrationTest/TestUtilities/IntegrationHelper.cs

```diff
@@ -202,172 +182,49 @@ public static void SetForegroundWindow(IntPtr window)
             NativeMethods.SwitchToThisWindow(activeWindow, true);
         }
 
-        public static void SendInput(NativeMethods.INPUT[] inputs)
+        public static void SendInput(KeyPress[] inputs)
         {
             // NOTE: This assumes that Visual Studio is the active foreground window.
+            var simulator = new InputSimulator();
 
-            LogKeyboardInputs(inputs);
-
-            var eventsInserted = NativeMethods.SendInput((uint)inputs.Length, inputs, NativeMethods.SizeOf_INPUT);
-
-            if (eventsInserted == 0)
-            {
-                var hresult = Marshal.GetHRForLastWin32Error();
-                throw new ExternalException("Sending input failed because input was blocked by another thread.", hresult);
-            }
-        }
+            ResetKeyState(VirtualKeyCode.LCONTROL);
```

Comment:
Before doing the reset, is it worth checking if the key is still active, so we can know if we're leaking modifier keys somewhere?

Reply:
I was not able to find a leak.